sdk/servicebus/Azure.Messaging.ServiceBus/src/Plugins/ServiceBusPlugin.cs

```diff
@@ -0,0 +1,221 @@
+        /// <param name="value">The value to set for the associated key.</param>
+#pragma warning disable CA1822 // Mark members as static
+        protected void SetUserProperty(ServiceBusReceivedMessage message, string key, object value)
+#pragma warning restore CA1822 // Mark members as static
+        {
+            message.SentMessage.Properties[key] = value;
+        }
+
+        /// <summary>
+        /// Sets the
+        /// <see cref="ServiceBusReceivedMessage.ContentType"/>.
+        /// </summary>
+        /// <param name="message">The message to modify.</param>
+        /// <param name="contentType">The content type to set on
+        /// the message.</param>
+
+#pragma warning disable CA1822 // Mark members as static
+        protected void SetContentType(ServiceBusReceivedMessage message, string contentType)
+#pragma warning restore CA1822 // Mark members as static
+        {
+            message.SentMessage.ContentType = contentType;
+        }
+
+        /// <summary>
+        /// Sets the <see cref="ServiceBusReceivedMessage.CorrelationId"/>.
+        /// </summary>
+        /// <param name="message">The message to modify.</param>
+        /// <param name="correlationId">The correlationId to set
+        /// on the message.</param>
+
+#pragma warning disable CA1822 // Mark members as static
+        protected void SetCorrelationId(ServiceBusReceivedMessage message, string correlationId)
+#pragma warning restore CA1822 // Mark members as static
+        {
+            message.SentMessage.CorrelationId = correlationId;
+        }
+
+        /// <summary>
+        /// Sets the <see cref="ServiceBusReceivedMessage.Label"/>.
+        /// </summary>
+        /// <param name="message">The message to modify.</param>
+        /// <param name="label">The label to set on the message.</param>
+
+#pragma warning disable CA1822 // Mark members as static
+        protected void SetLabel(ServiceBusReceivedMessage message, string label)
+#pragma warning restore CA1822 // Mark members as static
+        {
+            message.SentMessage.Label = label;
+        }
+
+        /// <summary>
+        /// Sets the <see cref="ServiceBusReceivedMessage.MessageId"/>.
+        /// </summary>
+        /// <param name="message">The message to modify.</param>
+        /// <param name="messageId">The message ID to set on the message.</param>
+
+#pragma warning disable CA1822 // Mark members as static
+        protected void SetMessageId(ServiceBusReceivedMessage message, string messageId)
+#pragma warning restore CA1822 // Mark members as static
+        {
+            message.SentMessage.MessageId = messageId;
+        }
+
+        /// <summary>
+        /// Sets the <see cref="ServiceBusReceivedMessage.PartitionKey"/>.
+        /// </summary>
+        /// <param name="message">The message to modify.</param>
+        /// <param name="partitionKey">The partion key to set on the message.</param>
+
+#pragma warning disable CA1822 // Mark members as static
```

Comment:
nit: Disable for the entire file?

Reply:
or just the section with all the Set<Property>() methods 🙂